scripts/updatebreaches.js

```diff
@@ -0,0 +1,97 @@
+"use strict";
+
+const got = require("got");
+
+const AppConstants = require("../app-constants");
+const HIBP = require("../hibp");
+
+
+const BREACHES_COLLECTION = "fxmonitor-breaches";
+const FX_RS_RECORDS = `${AppConstants.FX_REMOTE_SETTINGS_WRITER_SERVER}/buckets/main-workspace/collections/${BREACHES_COLLECTION}/records`;
+const FX_RS_COLLECTION = `${AppConstants.FX_REMOTE_SETTINGS_WRITER_SERVER}/buckets/main-workspace/collections/${BREACHES_COLLECTION}`;
+const FX_RS_PREVIEW_COLLECTION = `${AppConstants.FX_REMOTE_SETTINGS_WRITER_SERVER}/buckets/main-preview/collections/${BREACHES_COLLECTION}`;
+const FX_RS_USERNAME = AppConstants.FX_REMOTE_SETTINGS_USERNAME;
+const FX_RS_PASSWORD = AppConstants.FX_REMOTE_SETTINGS_PASSWORD;
+
+
+if (!FX_RS_USERNAME || !FX_RS_PASSWORD) {
+  console.error("updatebreaches requires FX_REMOTE_SETTINGS_USERNAME and FX_REMOTE_SETTINGS_PASSWORD.");
+  process.exitCode = 1;
+  return;
+}
+
+
+async function whichBreachesAreNotInRemoteSettingsYet(hibpBreaches) {
+  const remoteSettingsBreachesSet = new Set((await got(
+    `${AppConstants.FX_REMOTE_SETTINGS_ENDPOINT}/buckets/main/collections/${BREACHES_COLLECTION}/records`,
+    { json: true }
+  )).body.data.map(b => b.Name));
+
+  const newBreaches = [];
+  for (const breach of hibpBreaches) {
+    if (remoteSettingsBreachesSet.has(breach.Name)) {
+      continue;
+    }
+    newBreaches.push(breach);
+  }
+  return newBreaches;
+}
+
+
+(async () => {
+  const allHibpBreaches = await HIBP.req("/breaches");
+  const verifiedSiteBreaches = HIBP.filterBreaches(allHibpBreaches);
+
+  const newBreaches = await whichBreachesAreNotInRemoteSettingsYet(verifiedSiteBreaches);
+
+  if (!newBreaches) {
+    console.log("No new breaches found.");
+    return;
+  }
+
+  console.log(`${newBreaches.length} new breach(es) found.`);
+
+  const FX_RS_AUTH = Buffer.from(`${FX_RS_USERNAME}:${FX_RS_PASSWORD}`).toString("base64");
+  const remoteSettingsPreviewSet = new Set((await got(
+    FX_RS_PREVIEW_COLLECTION,
+    { json: true }
+  )).body.data.map(b => b.Name));
+
+  for (const breach of newBreaches) {
+    if (remoteSettingsPreviewSet.has(breach.Name)) {
+      console.log(`Breach "${breach.Name}" is already in Remote Settings, waiting for review.`);
+      continue;
+    }
+
+    const data = {
+      Name: breach.Name,
+      Domain: breach.Domain,
+      BreachDate: breach.BreachDate,
+      PwnCount: breach.PwnCount,
+      AddedDate: breach.AddedDate,
+    };
+
+    try {
+      // Create the record
+      await got.post(FX_RS_RECORDS, {
+        headers: {
+          "Content-Type": "application/json",
+          "authorization": `Basic ${FX_RS_AUTH}`,
```

Comment:
For `Basic` authentication to work, you'll need to ask for the creation of a Remote Settings account (eg. https://bugzilla.mozilla.org/show_bug.cgi?id=1512466).
If you plan on using your personal credentials to run this script, you'll have to use Bearer tokens (see [FAQ](https://remote-settings.readthedocs.io/en/latest/frequently-asked-questions.html#how-do-i-automate-the-publication-of-records))